frontend/src/components/pages/CampCreation/index.tsx

```diff
@@ -159,87 +170,254 @@ const CampCreationPage = (): React.ReactElement => {
+        setEndTime(editCamp.endTime);
+        setAgeLower(editCamp.ageLower);
+        setAgeUpper(editCamp.ageUpper);
+        setOffersEDLP(
+          editCamp.earlyDropoff.length !== 0 ||
+            editCamp.latePickup.length !== 0,
+        );
+        setEarliestDropOffTime(editCamp.earlyDropoff);
+        setLatestPickUpTime(editCamp.latePickup);
+        setPriceEDLP(editCamp.pickupFee);
+        setAddresLine1(editCamp.location.streetAddress1);
+        setAddresLine2(editCamp.location.streetAddress2 ?? "");
+        setCity(editCamp.location.city);
+        setProvince(editCamp.location.province);
+        setPostalCode(editCamp.location.postalCode);
+        setCampImageURL(editCamp.campPhotoUrl);
+
+        // Set camp level capacity from camp sessions
+        if (editCamp.campSessions.length > 0) {
+          // At the creation stage, all the sessions should have the same capacity
+          setCampCapacity(editCamp.campSessions[0].capacity);
+        }
+        setIsLoadedCampData(true);
+      }
+    };
+
+    // We only want to fetch the current state of the camp if we are editing a draft camp
+    if (editCampId) {
+      setInitialEditState();
+    }
+  }, [editCampId]);
+
+  const createUpdateCampHelper = async (
+    isPublishedCamp: boolean,
+    isNewCamp: boolean,
+  ): Promise<void> => {
+    const campFields: CreateUpdateCampRequest = {
+      active: isPublishedCamp,
+      ageLower,
+      ageUpper,
+      campCoordinators: [],
+      campCounsellors: [],
+      name: campName,
+      description: campDescription,
+      earlyDropoff: earliestDropOffTime,
+      endTime,
+      latePickup: latestPickUpTime,
+      location: {
+        streetAddress1: addressLine1,
+        streetAddress2: addressLine2,
+        city,
+        province,
+        postalCode,
+      },
+      startTime,
+      fee: dailyCampFee,
+      pickupFee: priceEDLP,
+      dropoffFee: priceEDLP,
+      formQuestions: customQuestions,
+      campSessions: scheduledSessions.map((campSession) => {
+        return {
+          dates: campSession.dates.map((date) => date.toString()),
+          capacity: campCapacity,
+        };
+      }),
+      volunteers: "",
+    };
+
+    try {
+      const campResponse = await createUpdateCamp(
+        campFields,
+        isNewCamp,
+        editCampId,
+      );
+
+      if (campResponse) {
+        history.push(CAMPS_PAGE);
+        toast({
+          description: `${campResponse.name} has been succesfully ${
+            isNewCamp ? "created" : "updated"
+          }`,
+          status: "success",
+          variant: "subtle",
+          duration: 3000,
+        });
+      } else {
+        toast({
+          description: `An error occurred with ${
+            isNewCamp ? "creating" : "updating"
+          } ${campFields.name}. Please try again.`,
+          status: "error",
+          variant: "subtle",
+          duration: 3000,
+        });
+      }
+    } catch (error: unknown) {
```

Comment:
Do we want the error toast to be more informative? I tried Save as draft and it failed, and the error is: "Error: You must fill out all the required fields in step 1 and schedule at least 1 session before creation a camp. Please try again." 

Feels like saving as draft should be allowed even without scheduling camp sessions though?

Reply:
that's a requirement we have because the camp list table uses the sessions to determine which year a camp belongs to. without any sessions, the camp would not show up in the camp list page on the staff tool. The clients already agreed to this requirement so we should be good

Reply:
So the create/edit call may throw an error with an message -- should that be displayed here? Could also be an alert if that's easier